taxonomy/api.py

```diff
@@ -0,0 +1,245 @@
+"""APIs for lore taxonomy application"""
+
+from django.db import transaction
+
+from taxonomy.models import (
+    Vocabulary,
+    Term,
+)
+from learningresources.models import (
+    LearningResource,
+    LearningResourceType,
+    Repository,
+)
+
+
+# pylint: disable=no-member
+def get_vocabulary(vocabulary_id):
+    """
+    Lookup vocabulary given its id
+
+    Args:
+        vocabulary_id (int): Vocabulary id
+
+    Returns:
+        Vocabulary: The vocabulary from the database
+    """
+    return Vocabulary.objects.get(id=vocabulary_id)
+
+
+# pylint: disable=too-many-arguments
+@transaction.atomic
+def create_vocabulary(
+        repository_id, name, description, required, vocabulary_type, weight
+):
+    """
+    Create a Vocabulary and save it in database
+
+    Args:
+        repository_id (int): Repository id
+        name (unicode): Name of Vocabulary
+        description (unicode): Description of Vocabulary
+        required (bool): Is Vocabulary required?
+        vocabulary_type (unicode): Vocabulary type
+        weight (int): Weight
+
+    Returns:
+        Vocabulary: The Vocabulary which was just added to the database
+
+    """
+
+    if vocabulary_type != "m" and vocabulary_type != "f":
+        raise TaxonomyAPIException("vocabulary_type must be either"
+                                   " 'm' (managed) or 'f' (free tagging)")
+
+    repository = Repository.objects.get(id=repository_id)
+
+    return Vocabulary.objects.create(
+        repository=repository,
```

Comment:
If the id is assigned directly without the database check the number may be invalid. Do you think that's a reasonable thing to check for here or should we rely on the caller already having validated that? 

Would it be more appropriate to not check and get a DoesNotExist error the next time Vocabulary.repository is called instead? It may be the more pythonic way, not sure